src/json_render.cpp

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright (c) 2018 Andrew Kelley
+ *
+ * This file is part of zig, which is MIT licensed.
+ * See http://opensource.org/licenses/MIT
+ */
+
+#include "analyze.hpp"
+#include "ast_render.hpp"
+#include "json_render.hpp"
+#include "os.hpp"
+
+#include <stdio.h>
+
+static const char *tld_id_str(TldId id) {
+    switch (id) {
+        case TldIdVar:       return "var";
+        case TldIdFn:        return "fn";
+        case TldIdContainer: return "container";
+        case TldIdCompTime:  return "comptime";
+    };
+    zig_unreachable();
+}
+
+static const char *type_table_entry_id_str(TypeTableEntryId id) {
```

Comment:
I think you maybe want to use
```c
const char *type_id_name(TypeTableEntryId id);
```
instead of this?

Reply:
Done